wallet/types/src/account_info.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common::address::pubkeyhash::PublicKeyHash;
+use common::chain::{OutPoint, Transaction};
+use common::primitives::Id;
+use crypto::key::extended::ExtendedPublicKey;
+use crypto::key::hdkd::child_number::ChildNumber;
+use crypto::key::hdkd::derivation_path::DerivationPath;
+use serialization::{Decode, Encode};
+use storage::HasPrefix;
+
+/// The account id is described by it's public key hash or a random value for
+/// non deterministic accounts
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Encode, Decode)]
+pub struct AccountId(PublicKeyHash);
+
+impl AccountId {
+    pub fn new_from_xpub(pub_key: &ExtendedPublicKey) -> Self {
+        let pk = pub_key.clone().into_public_key();
```

Comment:
I think you can add a method `public_key` method that returns a reference to `ExtendedPublicKey` to get rid of this clone.

Reply:
I tried that but the `ExtendedPublicKey` holds `ExtendedPublicKeyHolder::Secp256k1Schnorr(Secp256k1ExtendedPublicKey)` and in turn holds a `Secp256k1PublicKey`. But we need a `PublicKey` so a reference isn't possible as far as I can understand.